File: src/main/java/net/matees/arcade/mobrush/settings/RandomMobCount.java
package net.matees.arcade.mobrush.settings;

import org.bukkit.Material;

import net.matees.settings.BooleanSetting;

public class RandomMobCount extends BooleanSetting {

    public Boolean randomMobCount = true;

    private static final RandomMobCount INSTANCE = new RandomMobCount();

    private RandomMobCount() {

    }

    public static RandomMobCount getInstance() {
        return INSTANCE;
    }

    @Override
    public String getName() {
        return "Random Mob Count";
    }

    @Override
    public String getDescription() {
        return "Toggle random amount of mobs spawned";
    }

    @Override
    public Boolean getSetting() {
        return this.randomMobCount;
    }

    @Override
    public void setSetting(Boolean setting) {
        this.randomMobCount = setting;
    }

    @Override
    public Material getMenuItemMaterial() {
        return Material.REDSTONE_BLOCK;
    }

    @Override
    public int getMenuItemSlot() {
        return 11;
    }

}

File: src/main/java/net/matees/arcade/mobrush/settings/RandomMobCount.kt
package net.matees.arcade.mobrush.settings

import net.matees.settings.BooleanSetting
import org.bukkit.Material

class RandomMobCount private constructor() : BooleanSetting() {
    private var randomMobCount: Boolean = true

    override val name: String
        get() = "Random Mob Count"

    override val description: String
        get() = "Toggle random amount of mobs spawned"

    override var setting: Boolean?
        get() = this.randomMobCount
        set(setting) {
            if (setting != null) {
                this.randomMobCount = setting
            }
        }

    override val menuItemMaterial: Material
        get() = Material.REDSTONE_BLOCK

    override val menuItemSlot: Int
        get() = 11

    companion object {
        val instance: RandomMobCount = RandomMobCount()
    }
}
